src/pages/Transfer/CrossChainTransferForm/index.tsx

```diff
@@ -1,52 +1,187 @@
+  const [approxUsdValue, setApproxUsdValue] = React.useState<string>('0');
+
+  const dispatch = useDispatch();
   const { t } = useTranslation();
 
   const {
-    collateralTokenTransferableBalance
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<CrossChainTransferFormData>({
+    mode: 'onChange'
+  });
+
+  const {
+    parachainStatus,
+    address,
+    prices
   } = useSelector((state: StoreType) => state.general);
 
+  const onSubmit = async (data: CrossChainTransferFormData) => {
+    if (!address) return;
+    if (!destination) return;
+
+    try {
+      setSubmitStatus(STATUSES.PENDING);
+
+      await xcmTransfer(
+        address,
+        destination.address,
+        newMonetaryAmount(data[TRANSFER_AMOUNT], COLLATERAL_TOKEN, true)
+      );
+
+      setSubmitStatus(STATUSES.RESOLVED);
+    } catch (error) {
+      setSubmitStatus(STATUSES.REJECTED);
+      setSubmitError(error);
+    }
+  };
+
+  const handleConfirmClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!address) {
+      dispatch(showAccountModalAction(true));
+      event.preventDefault();
+    }
+  };
+
+  // This ensures that triggering the notification and clearing
+  // the form happen at the same time.
+  React.useEffect(() => {
+    if (submitStatus !== STATUSES.RESOLVED) return;
+
+    toast.success(t('transfer_page.successfully_transferred'));
+
+    reset({
+      [TRANSFER_AMOUNT]: ''
+    });
+  }, [
+    submitStatus,
+    reset,
+    t
+  ]);
+
+  const handleValueChange = (event: any) => {
+    const value = newMonetaryAmount(event.target.value, COLLATERAL_TOKEN, true);
+    const usd = getUsdAmount(value, prices.governanceToken.usd);
+
+    setApproxUsdValue(usd);
+  };
+
   return (
-    <form className='space-y-8'>
-      <FormTitle>
-        {t('transfer_page.cross_chain_transfer_form.title')}
-      </FormTitle>
-      <div>
-        <AvailableBalanceUI
-          label={t('transfer_page.cross_chain_transfer_form.balance')}
-          balance={displayMonetaryAmount(collateralTokenTransferableBalance)} />
+    <>
+      <form
+        className='space-y-8'
+        onSubmit={handleSubmit(onSubmit)}>
+        <FormTitle>
+          {t('transfer_page.cross_chain_transfer_form.title')}
+        </FormTitle>
+        <div>
+          <div>
+            <TokenField
+              onChange={handleValueChange}
+              id={TRANSFER_AMOUNT}
+              name={TRANSFER_AMOUNT}
+              ref={register({
+                required: {
+                  value: true,
+                  message: t('transfer_page.cross_chain_transfer_form.please_enter_amount')
+                }
+              })}
+              error={!!errors[TRANSFER_AMOUNT]}
+              helperText={errors[TRANSFER_AMOUNT]?.message}
+              label={COLLATERAL_TOKEN_SYMBOL}
+              approxUSD={`≈ $ ${approxUsdValue}`} />
+          </div>
+        </div>
+        <div>
```

Comment:
@tomjeatt 
I think these divs are redundant. I tested it by removing them. Yes, there's a problem with the layout without these divs but we can concretely resolve it by wrapping with div inside `AccountSelector` and `ChainSelector`.

If we need `Chains` and `Accounts` in other places, then we may have to use these divs again, which is not good.